### PR TITLE
Add JavaScript Bundling feature using oax

### PR DESCRIPTION
**Pull request by JakeChampion**
oax is the bundling tool we are using, it is a binary dependency that we include as an npm package.

I've included integration/end-to-end tests for the v3/bundle/js endpoint.

All errors that occur in the /v3/bundle/js endpoint are surfaced to the user as a JavaScript error. E.G. If they requested a component which does not exist, we return a JavaScript response similar to 

```js
throw new Error("Sorry, the component 'o-marquee' does not exist")
```

### CODE REVIEW

File: functions/jsBundle.js [hunk 1]
```diff
@@ -7,10 +7,9 @@ const { jsBundle } = require("../src/jsBundle");
 
 const jsHandler = RavenLambdaWrapper.handler(Raven, async event => {
   try {
-    return jsBundle(event.queryStringParameters);
+    return await jsBundle(event.queryStringParameters);
```
Comment (JakeChampion) [inline functions/jsBundle.js]: This await is required to make the Promise rejection from jsBundle be turned into a thrown error that we can catch.

Comment (chee) [reply functions/jsBundle.js]: worth a comment in the file?

Comment (JakeChampion) [reply functions/jsBundle.js]: Good idea, I've added one 👍

File: package.json [hunk 2]
```diff
@@ -4,52 +4,54 @@
...
     "debug": "^4.1.1",
     "decompress": "^4.2.0",
     "directory-exists": "^2.0.1",
-    "execa": "^2.0.5",
+    "execa": "^3.2.0",
     "http-errors": "^1.7.3",
     "immutable": "^4.0.0-rc.12",
     "lodash": "^4.17.15",
-    "object.fromentries": "^2.0.0",
+    "oax": "^0.5.24",
```
Comment (chee) [inline package.json]: 😊

File: functions/jsBundle.js [hunk 3]
```diff
@@ -20,9 +21,29 @@ const jsHandler = RavenLambdaWrapper.handler(Raven, async event => {
...
+          headers: {
+            "content-type": "application/javascript;charset=UTF-8",
+            "cache-control": "max-age=0, must-revalidate, no-cache, no-store",
+          },
+        },
+      );
+    } else {
+      // TODO: output stacktrace and error message if not running in production
+      return createError.InternalServerError(
+        `throw new Error(${JSON.stringify("Could not create bundle")})`,
```
Comment (chee) [inline functions/jsBundle.js]: should we include something like `Origami Build Service returned an error:` in the message to make debugging easier when you see it in the console?

Comment (JakeChampion) [reply functions/jsBundle.js]: That's a good idea!

File: src/jsBundle.js [hunk 4]
```diff
@@ -2,53 +2,62 @@
...
+const { SolveFailure } = require("./modules/SolveFailure");
+const { UserException, FormatException } = require("./modules/HOME");
 
-console.clear();
 const jsBundle = async (querystring = {}) => {
   await fs.mkdir("/tmp/bundle/", { recursive: true });
   const bundleLocation = await fs.mkdtemp("/tmp/bundle/");
 
   try {
-    await rmrf(`/Users/jake.champion/.jake-cache`);
```
Comment (chee) [inline src/jsBundle.js]: oh no, but that's where i was keeping my local jakes

Comment (JakeChampion) [reply src/jsBundle.js]: 😂 Sorry Chee

File: functions/jsBundle.js (hunk 3 above)
Comment (JakeChampion) [reply functions/jsBundle.js]: Done! ca751a511d1a10da36b6855892075cb8f0f526b3

**Pull request merged**